File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.py
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        result = []
    
        for index,row in enumerate(mat):
            soldier = (sum(row) ,index)
            result.append(soldier)

        result.sort()     

        return [index for _,index in result[:k]]

File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.ts
function kWeakestRows(mat: number[][], k: number): number[] {
    
      const result = [];

      let rowIndex = 0;
      for (const row of mat) {
        let counterSoldier = row.reduce(
          (accomulator, currentValue) => accomulator + currentValue,
          0
        );
        const soldier = [counterSoldier, rowIndex];
        result.push(soldier);
        rowIndex++;
      }

      result.sort((a, b) => a[0] - b[0]);

      return result.slice(0, k).map((rowIndex) => rowIndex[1]);
    
};